### PR TITLE
feat(build): add mage dagger:run test:migration

### PR DESCRIPTION
**Pull request by GeorgeMac**
Fixes FLI-376

Adds a new mage target: `mage dagger:run test:migration`.
The test ensures that any new migrations are compatible and covers the safety of the upgrade path.

The test does the following:
1. Pulls `flipt/flipt:latest`.
2. Runs it against a SQLite DB file (including migrations).
3. Imports the read-only testdata.
4. Builds Flipt from local state of project.
5. Runs `flipt migrate` using new edge build over DB created in (2) and populated in (3).
6. Runs edge Flipt over the migrated database.
7. Runs the readonly test suite against instance in (6).

## Example

I wrote a migration to `ALTER TABLE flags DROP COLUMN description` and then ran this test:

```
┃ === RUN   TestReadOnly/[Protocol_"grpc";_Namespace_"default";_Authentication_false]/GetFlag
┃     readonly_test.go:69:
┃             Error Trace:    /src/build/testing/integration/readonly/readonly_test.go:69
┃             Error:          Received unexpected error:
┃                             rpc error: code = Internal desc = no such column: description
┃             Test:           TestReadOnly/[Protocol_"grpc";_Namespace_"default";_Authentication_false]/GetFlag
┃ === RUN   TestReadOnly/[Protocol_"grpc";_Namespace_"default";_Authentication_false]/ListFlags
┃     readonly_test.go:86:
┃             Error Trace:    /src/build/testing/integration/readonly/readonly_test.go:86
┃             Error:          Received unexpected error:
┃                             rpc error: code = Internal desc = no such column: description
┃             Test:           TestReadOnly/[Protocol_"grpc";_Namespace_"default";_Authentication_false]/ListFlags
```

This is not the most representative test, as I believe it would fail our other tests too.
But it at-least demonstrates that it picks up these kinds of failure at this stage.

## TODO

- [x] Install this into a GH action
- [x] Force it to fail locally (going to add a migration to drop a column)

### CODE REVIEW

**Review by markphelps — APPROVED** [review]
ty! ❤️

File: build/testing/migration.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		WithServiceBinding("flipt", latest.WithExec(nil)).
+		WithExec([]string{"/flipt", "import", "--address", "grpc://flipt:9000", "import.yaml"}).
+		Sync(ctx)
+	if err != nil {
+		return err
+	}
+
+	// run migration with edge Flipt build
+	flipt, err = flipt.
+		// WithEnvVariable("FLIPT_LOG_FILE", "/var/log/flipt/output.txt").
```
Comment (yquansah) [inline build/testing/migration.go:43]: Is this comment intentional?

**Review by yquansah — APPROVED** [review]
Nice!

File: build/testing/migration.go (hunk 1 above)
Comment (GeorgeMac) [reply build/testing/migration.go:43]: Ahh no, good spot, I was using this to reduce the log output volume at one point during debugging.

Comment (GeorgeMac) [inline build/testing/migration.go:43]:
```suggestion
```

**Pull request merged**